uselotus/lotus
# Task : Show events and properties preview on Create Metric Dialog

Issue : https://github.com/uselotus/lotus/issues/577

Current Behavior : Create Metric Dialog allows user to create a new metric and define fields like Event Name and Property Name for Guage Type.

Expected Behavior: 
1. Change simple Input fields to Select field, list of existing events are shown to the user.
2. List of events sent by the backend is distinct by event names.
3. Upon selecting a event, Property Name field reflects upon showing possible properties event has. 

Test Feature =>
1. Login in using demo3 credentials
2. headon over to Metrics tabs. 
3. notice all the events tracking on main view.
4. click "Create Metric" button to open up dialog.
5. navigate over to event name and property name field to notice those events showing up. 

Todo steps =>
Currently feature is using Select fields with prop mode="tags" which gives user bad experience about field being multi select. And antd does not have a component specific to our needs. 
1. Implement a common component per design needs, Single Select Field


Screenshots =>

![Screenshot 2023-03-01 at 5 24 54 PM](https://user-images.githubusercontent.com/6052826/222307012-3dd3203f-dc45-47b5-816d-ab2ae995d434.jpeg)
![Screenshot 2023-03-01 at 5 25 05 PM](https://user-images.githubusercontent.com/6052826/222307013-7b1e7e60-aa10-484e-8472-968a2d78c7e6.jpeg)
![Screenshot 2023-03-01 at 5 25 12 PM](https://user-images.githubusercontent.com/6052826/222307016-053d55a8-4cec-464f-a812-62f1772c0479.jpeg)
